src/test/testing_common.c

```diff
@@ -89,6 +89,22 @@ setup_directory(void)
                  (int)getpid(), rnd32);
     r = mkdir(temp_dir);
   }
+#elif defined(__ANDROID__)
+  /* tor might not like the default perms, so create a subdir */
+  tor_snprintf(temp_dir, sizeof(temp_dir),
+               "/data/local/tmp/tor_%d_%d_%s",
+               (int) getuid(), (int) getpid(), rnd32);
+  r = mkdir(temp_dir, 6770);
+  if (r) {
+    fprintf(stderr, "Can't create directory %s:", temp_dir);
+    perror("");
+    exit(1);
+  }
+  tor_snprintf(temp_dir, sizeof(temp_dir),
```

Comment:
I'm not sure why this secondary directory is needed.  You mention permissions in the commit message, but this is created with the same permissions as the parent directory.

Reply:
* _/data/local/tmp_ is not _/tmp_, it works differently, including different perms
* _/data/local/tmp_ does not get automatically wiped, only manually
* having the first dir with test subdirs keeps things a bit neater after multiple runs